docs/value.md

```diff
@@ -0,0 +1,236 @@
+
+
+# mapbox::base values
+
+Values are used widely in Mapbox libraries, for instance when working with geometries or data in JSON format. They can be nested to create complex structures.
```

Comment:
Can you please add a sentence on why we use value types? Something like _we exchange strong typing in favor of flexibility, reducing API surface and simplifying ergonomics_.